ed25519/src/lib.rs

```diff
@@ -0,0 +1,11 @@
+#[cfg(feature = "hacl")]
+pub mod hacl {
```

Comment:
We shouldn't expose any underlying names. The caller doesn't care where the code may come from. That was my last bullet point in the previous review. This is again for all crates.
The crate should expose one libcrux API, independent of the implementation(s) underneath.
You could solve that by renaming things, I don't think there's the need to change anything else.

Reply:
We need to in some cases, because e.g. the hmac crate needs access to sha2. But yes, I can drop it where we don't specifically need it.

Reply:
Right. I think we should have a good way of marking internal APIs that we only expose because another libcrux crate needs them. Maybe putting them behind an additional feature is the easiest.
But let's file a follow up for that and merge this as is for now.